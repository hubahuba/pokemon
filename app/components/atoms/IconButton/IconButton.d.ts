import {TouchableOpacityProps} from 'react-native/Libraries/Components/Touchable/TouchableOpacity';

export interface IconButtonProps extends TouchableOpacityProps {
  iconName: string;
  color?: string;
}
