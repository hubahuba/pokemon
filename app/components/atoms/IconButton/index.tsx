import * as React from 'react';
import {TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/AntDesign';
import {IconButtonProps} from '~atoms/IconButton/IconButton';

export default function IconButton(props: Readonly<IconButtonProps>) {
  const {iconName, color, className, ...touchOpacityProps} = props;
  return (
    <TouchableOpacity className={className} {...touchOpacityProps}>
      <Icon color={color} name={iconName} size={24} />
    </TouchableOpacity>
  );
}
