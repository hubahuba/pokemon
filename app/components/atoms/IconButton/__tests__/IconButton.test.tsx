import React from 'react';
import renderer from 'react-test-renderer';
import IconButton from '../index';

it('renders Card correctly', () => {
  const tree = renderer.create(<IconButton iconName="close" />).toJSON();
  expect(tree).toMatchSnapshot();
});
