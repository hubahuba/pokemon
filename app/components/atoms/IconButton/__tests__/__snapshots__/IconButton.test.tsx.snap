// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 24,
        },
        undefined,
        {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    î˜³
  </Text>
</View>
`;
